### ISSUE TITLE
TextInput on working with custom fonts

### ISSUE THREAD

**Issue by benjamin-r-erb** [nandorojo/dripsy#64]
Hi @nandorojo, love the project -- thanks for making this package available. 

I'm experiencing issues when using custom fonts specifically when using the dripsy imported TextInput. All my other pages and Text items are using the custom font appropriately.

Have you experienced this or aware of what might be causing it? If I replace the dripsy TextInput with the standard react-native the error goes away.

Thanks!!
Ben

```
AppEntry.bundle?plat…&minify=false:28030 fontFamily "NunitoSans" is not a system font and has not been loaded through Font.loadAsync.

- If you intended to use a system font, make sure you typed the name correctly and that it is supported by your device operating system.

- If this is a custom font, be sure to load it with Font.loadAsync. 
```

**Comment by nandorojo** [comment]
I don't think I have any issues with my inputs. I'll check to make sure.

**Comment by nandorojo** [comment]
Can I see your theme?

**Comment by benjamin-r-erb** [comment]
Thanks @nandorojo !!

Here's the component I'm using it in (still a work in progress, but curious if anything stands out):

```
import React from 'react';
//import {TextInput} from 'react-native';
import {View, Text, TextInput} from 'dripsy';
import {useField} from 'formik';

export default function FormTextInput({fieldName, label, labelSX, errorSX, ...props}) {
  const [field, meta] = useField(fieldName);
  let showTitle = meta.touched && Object(field.value).length > 0
  let showError = meta.error && meta.touched
  return (
    <View {...props}> 
      {showTitle ? <Text sx={labelSX}>{label}</Text> : <Text />}
      <TextInput
        value={field.value}
        placeholder={label}
        placeholderTextColor='gray'
        onChangeText={field.onChange(fieldName)}
        onBlur={field.onBlur(fieldName)}
      />
      {showError ? <Text style={errorSX ?? {color: 'red'}}>{meta.error}</Text> : <Text />}
    </View>
  );
}
```

And here is my App.js with the theme:

```
import "react-native-gesture-handler";
import React from "react";
import NavigationRoot from "./NavigationRoot";
import { Provider as AuthProvider } from "./src/context/AuthContext";
import { DripsyProvider } from "dripsy";
import Amplify from "@aws-amplify/core";
import { Root, RootStore } from "./src/stores/mobxStore";
import Fonts from "./src/constants/fonts";
import _ from "lodash";
import awsconfig from "./aws-exports";

Amplify.configure(awsconfig);

const App = ({}) => {
  const theme = {
    space: [0, 4, 8, 16, 32, 64, 128, 256, 512],
    customFonts: {
      NunitoSans: {
        bold: "NunitoSans_700Bold",
        default: "NunitoSans_400Regular",
        normal: "NunitoSans_400Regular",
        '400': "NunitoSans_400Regular",
        '600': "NunitoSans_600SemiBold",
        '700': "NunitoSans_700Bold",
        '800': "NunitoSans_800ExtraBold",
        '900': "NunitoSans_900Black",
      },
    },
    fonts: {
      root: "NunitoSans",
      // heading: 'NunitoSans_700Bold',
      // monospace: 'Menlo, monospace',
    },
    fontSizes: [12, 14, 16, 20, 24, 32, 48, 64, 96],
    fontWeights: {
      body: "400",
      semibold: "600",
      heading: "700",
      bold: "800",
    },
    lineHeights: {
      body: 1.5,
      heading: 1.125,
    },
    colors: {
      text: "#000000",
      background: "#ffffff",
      primary: "#95a792",
      secondary: "#596c68",
      accent: "#0F3287",
      highlight: "#e3d9ca",
      muted: "#f0ece2",
      link: "#0091FF",
      gray: "#979797",
      modes: {
        dark: {
          text: "#fff",
          background: "#000",
          primary: "#5E8C82",
          secondary: "#95a792",
          muted: "#111",
        },
      },
    },
    text: {
      body: {
        fontWeight: "body",
        // this edits all non-styled <Text> components
      },
      headerText: {
        //fontFamily: 'NunitoSans_800ExtraBold',
        fontWeight: "bold",
        fontSize: [4, 4, 5],
        textAlign: "center",
      },
      titleText: {
        //fontFamily: 'NunitoSans_800ExtraBold',
        fontWeight: "semibold",
        fontSize: [2, 2, 3],
        textAlign: "center",
      },
      labelText: {
        fontSize: [2, 2, 3],
        fontWeight: "semibold",
      },
      bodyText: {
        fontWeight: "body",
        fontSize: [2, 2, 3],
      },
      boldText: {
        fontWeight: "bold",
        fontSize: [2, 2, 3],
      },
      linkText: {
        fontWeight: "semibold",
        fontSize: [2, 2, 3],
        color: "link",
      },
    },
    styles: {
      root: {
        fontFamily: "body",
        lineHeight: "body",
        fontWeight: "body",
      },
      code: {
        // fontFamily: 'monospace',
        // fontSize: 'inherit',
      },
    },
    buttons: {
      primary: {
        // you can reference other values defined in the theme
        color: "white",
        bg: "secondary",
      },
      secondary: {
        color: "text",
        bg: "muted",
      },
      accent: {
        color: "white",
        bg: "accent",
      },
    },
    variants: {
      button: {
        outline: {},
        normal: {},
      },
    },
  };

  const root = new Root();

  return (
    <Fonts>
      <RootStore.Provider value={root}>
        <AuthProvider>
          <DripsyProvider theme={theme}>
            <NavigationRoot />
          </DripsyProvider>
        </AuthProvider>
      </RootStore.Provider>
    </Fonts>
  );
};

export default App;

```

**Comment by nandorojo** [comment]
Okay this all looks good, I have two ideas for where the problem might be. I think it's going to be with `expo-font`. Could I see your expo font loader file?

PS if you put `jsx` after the first three back-ticks in the code you share, it formats the code with nice colors

Like this:

```jsx

**Comment by benjamin-r-erb** [comment]
@nandorojo Thanks...that's helpful. See below

```jsx
import React from "react";
import { useFonts } from 'expo-font';
import AppLoading from "expo-app-loading";

export default function Fonts({ children }: { children: React.ReactNode }) {
  const [loaded] = useFonts({
    // 🚨🚨🚨 the name (`sans`) of the default weight here should equal the key from theme.customFonts!
    // otherwise, you will need to explicitly set the fontWeight everywhere
    // since we have theme.customFonts.sans, we name this `sans`
    // ["sans"]: require("./public/fonts/sansBook.ttf"),
    // ["sansBold"]: require("./public/fonts/arialBlack.ttf"),

    // same goes here, load in the default font name with the one that matches your theme.customFonts
    // ["arial"]: require("./public/fonts/arialBook.ttf"),
    // ["arialBold"]: require("./public/fonts/arialBold.ttf"),
  

    ["NunitoSans_900Black"]: require("../../assets/Nunito_Sans/NunitoSans-Black.ttf"),
    ["NunitoSans_900Black_Italic"]: require("../../assets/Nunito_Sans/NunitoSans-BlackItalic.ttf"),
   
    ["NunitoSans_700Bold"]: require("../../assets/Nunito_Sans/NunitoSans-Bold.ttf"),
    ["NunitoSans_700Bold_Italic"]: require("../../assets/Nunito_Sans/NunitoSans-BoldItalic.ttf"),

    ["NunitoSans_800ExtraBold"]: require("../../assets/Nunito_Sans/NunitoSans-ExtraBold.ttf"),
    ["NunitoSans_800ExtraBold_Italic"]: require("../../assets/Nunito_Sans/NunitoSans-ExtraBoldItalic.ttf"),

    ["NunitoSans_200ExtraLight"]: require("../../assets/Nunito_Sans/NunitoSans-ExtraLight.ttf"),
    ["NunitoSans_200ExtraLight_Italic"]: require("../../assets/Nunito_Sans/NunitoSans-ExtraLightItalic.ttf"),

    ["NunitoSans_400Regular_Italic"]: require("../../assets/Nunito_Sans/NunitoSans-Italic.ttf"),
    ["NunitoSans_300Light"]: require("../../assets/Nunito_Sans/NunitoSans-Light.ttf"),

    ["NunitoSans_300Light_Italic"]: require("../../assets/Nunito_Sans/NunitoSans-LightItalic.ttf"),
    ["NunitoSans_400Regular"]: require("../../assets/Nunito_Sans/NunitoSans-Regular.ttf"),

    ["NunitoSans_600SemiBold"]: require("../../assets/Nunito_Sans/NunitoSans-SemiBold.ttf"),
    ["NunitoSans_600SemiBold_Italic"]: require("../../assets/Nunito_Sans/NunitoSans-SemiBoldItalic.ttf"),
 
  });

  if (!loaded) return <AppLoading />;

  return <>{children}</>;
}
```

**Comment by nandorojo** [comment]
Notice the comments I left there. They aren't being followed. 

The key should be `NunitoSans` for the default font weight, not `NunitoSans_400Regular`. Then, your `theme.customFonts.NunitoSans[400]` (sams with `default` and `normal`) should be `NunitoSans`.

**Comment by nandorojo** [comment]
Any update?

**Comment by benjamin-r-erb** [comment]
@nandorojo Thanks again for following up on this. We made the changes you suggested in the loading to set the default weight to the generic font name 'NunitoSans' and that got rid of the error that was only appearing for us in the TextInput.

We also took a closer look at the new instructions. The font we are using has many different files for each style/weight, so the arial example with multiple weights per font name caused some confusion.

The other huge realization was that we had an unnecessary theme.text.body style element that was causing our font styling not to work unless using a text variant (which we had adopted as a workaround).

Thankfully everything is working now and we understand the process much better.

Thank you so much for your help and for making this awesome library available!

Best,
Ben
